### PR TITLE
Use foreman and puma to run web server

### PR DESCRIPTION
**Pull request by dwhenry**

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env puma
+
+rackup DefaultRackup
+
+environment ENV['RACK_ENV'] || 'development'
```
Comment (benlovell) [inline config/puma.rb]: Bringing this over here from #6

Comment (dwhenry) [inline config/puma.rb]: I know it doesn't make a lot of difference by fetch is slightly slower - or at least used to be in ruby 2.2:

https://github.com/JuanitoFatas/fast-ruby#hash-vs-hashfetch-code

Comment (benlovell) [inline config/puma.rb]:
```ruby
require 'benchmark/ips'

hash = Hash.new

Benchmark.ips do |b|
  b.report('normal') do
    hash[:missing] || 'hi'
  end

  b.report('fetch') do
    hash.fetch(:missing, 'hi')
  end
end
```

```
hurrdurr~/code <master> # ruby fetch.rb
Warming up --------------------------------------
              normal   126.465k i/100ms
               fetch   126.291k i/100ms
Calculating -------------------------------------
              normal      6.699M (± 6.4%) i/s -     33.387M
               fetch      6.506M (± 6.7%) i/s -     32.457M
```

Negligible in 2.3.0. I don't think performance is a concern here.

Comment (benbarnett) [inline config/puma.rb]: Yeah we've used `fetch` in [other](https://github.com/alphagov/govspeak/blob/master/lib/govspeak/header_extractor.rb#L22) [places](https://github.com/guidance-guarantee-programme/pension_guidance/blob/master/spec/controllers/concerns/breadcrumbs_controller_concern_spec.rb#L12) too so probably best to stick with that style.

Comment (dwhenry) [inline config/puma.rb]: done

**Comment by benlovell** [comment]
🚢 🚀 :shipit: Nice.

**Pull request merged**